src/app.service.ts

```diff
@@ -9,72 +9,67 @@ import { date, number, string } from 'joi';
 import axios from 'axios';
 import { config } from 'rxjs';
 import Joi from "joi";
+import { Session } from 'express-session';
 
 @Injectable()
 export class AppService {
-  public transactions: TransactionModel[] =[]
-  public payUnitData={
-    transaction_id:10,
-    transaction_amount:10000,
-    currency:'USD',
-    return_url:'http://localhost:3000',
-  };
-  // private all
-  private completeTransaction = {
-    gateway: 'mtnmomo',
-    transaction_amount: 1001,
-    transaction_id: 1005,
-    phone_number: 681479697,
-    currency: 'USD',
-    paymentType: 'button',
-  };
+  private all
+  public  getIninitial: GetTransaction[] =[]
+  public transaction: TransactionModel[] =[]
+  public  payment:Payment[] =[]
   private urlinitialized = 'https://app.payunit.net/api/gateway/initialize';
   private paymentUrl = 'https://app.payunit.net/api/gateway/makepayment';
-  private urlproceed = 'https://app.payunit.net/api';
+  private urlproceed = 'https://app.payunit.net/api/gateway/gateways?t_url=""&t_id=""&t_sum=""';
```

Comment:
Put the base URL in the environment file to avoid initializing a lot of variables as you just did and just call it when you need it and append the route to it. The base URL here will be https://app.payunit.net/api and the rest after it is called routes.